lib/imgAutosize.js

```diff
@@ -0,0 +1,91 @@
+const url = require('url');
+const path = require('path');
+const util = require('util');
+const http = require('http');
+
+const imageSize = require('image-size');
+const imageSizeAsPromise = util.promisify(imageSize);
+
+module.exports = {
+    default: function(options) {
+        options = options || {};
+        options.root = options.root || '.';
+        options.processEmptySize = options.processEmptySize || false;
+
+        const projectRoot = path.resolve(options.root);
+
+        return function(tree) {
+            const htmlRelativePath = path.relative(process.cwd(), path.dirname(tree.options.to || ''));
+
+            const promises = [];
+
+            tree.match({ tag: 'img' }, img => {
+                img.attrs = img.attrs || {};
+
+                const imagePath = translatePath(projectRoot, htmlRelativePath, img.attrs.src);
+
+                const updateWidth =
+                    img.attrs.width == 'auto' ? true :
+                    !img.attrs.width ? options.processEmptySize : false;
+
+                const updateHeight =
+                    img.attrs.height == 'auto' ? true :
+                    !img.attrs.height ? options.processEmptySize : false;
```

Comment:
A double ternary operator is hard to understand. Can you replace it with something more readable (e.g. multiple ifs)?